### DISCUSSION TITLE
An unfortunate menu item name, which led to the setting of an inappropriate default state of the variable for Suppress

### DISCUSSION THREAD

**Discussion by belinski-roman** [FreeCAD/FreeCAD#18722] [Discussions that will turn in to issues but need refining]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

PR #13578 added an item to the context menu that is responsible for suppressing an element in the tree. This is a very good change, but there is some inconsistency in the interface and behavior. 

Currently, activating suppression means checking a box (this works great for selecting an active Body, but not for our case) and is called Suppressed, which is slightly confusing (since the initial visibility of the element does not imply that this is necessarily its starting state). A part with several configurations (for example, a plate with a different number of holes) in its basic version may in principle not contain any cutouts (but they may be present in other versions of this part). 

I suggest several steps to eliminate this situation:

1. Rename Suppressed to Toggle Suppressed (which will become uniform by analogy with Toggle Freeze, Toggle Visibility and Toggle Transparency). I took the liberty to depict a possible icon, but this is a job for the design group.

![Рис 1](https://github.com/user-attachments/assets/93f3434a-fb0b-4f8e-bf77-199ecb5b8746)

2. Invert the assignment of the variable c true/false (suppressed/no suppressed) to the opposite, where true is no suppressed and false is suppressed (by analogy with the Visibility property). This is more logical because true is one (the presence of something), and false is zero (absence, emptiness). This change may need to be backported to version 1.0.

![Рис 2](https://github.com/user-attachments/assets/d4d49ef8-8c15-449c-afc8-0c8300d0c8a8)

There may be a side benefit to my request. In the request #18473, it is proposed to allow the Occurrence parameter to be set to zero when creating LinerPattern and PolarPattern.  The author of this PR suggests the possibility of automatically suppressing the array in the tree when Occurrences=0. Together with my PR, this would be easy to achieve by associating the Occurrences parameter of LinerPattern and PolarPattern with their suppression state. And since values >1 will still be taken as true, this should work fine. 

As an example, consider a part that has four execution options and each option will be controlled only using one VarSet property (let's call it Instances). Let's choose as the type of variable. App::PropertyFloat (for properties that have a status switch, you can create an expression with a link not only to the App::PropertyBool (true/false), but also on the App::PropertyFloat (1/0), which is what we actually use in the example) and assign it the value 1.

![Рис 3](https://github.com/user-attachments/assets/0ec205f6-ce95-4026-a7b6-996a8f57ed9d)

Next, create a rectangular plate and use the Pocket command to make one round hole first and then multiply it using a LinerPattern. We will associate the previously created property with the Suppress property of our hole and associate it with the Occurrences parameter of our LinerPattern.

![Рис 4](https://github.com/user-attachments/assets/671a0287-3c95-4acc-9044-23b0f5815905)

The result of these actions is shown in the following illustration:

![Рис 5](https://github.com/user-attachments/assets/b7f18c53-f45d-4e43-9298-4e8aa244259e)

### Full version info

```shell
OS: Windows 10 build 19044
Architecture: x86_64
Version: 1.1.0dev.39708 (Git) Conda
Build type: Release
Branch: main
Hash: 173ff4da1ef4f075d56c0dc617d1bf00d5f49d4b
Python 3.11.11, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: Russian/Russia (ru_RU)
Stylesheet/Theme/QtStyle: FreeCAD Light.qss/FreeCAD Light/Fusion
```

### Subproject(s) affected?

PartDesign

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
For discussion @FreeCAD/design-working-group 
IMO the menu item with string "Toggle" is always worse than the current design where you instantly see the current state and can switch between suppressed on/off, this was discussed when the feature was introduced and also switched for other commands (like active part/body).
I don't see a benefit of setting suppressed to true when false is the default state, as each object is by default not suppressed. Changing the label to NotSuppressed = true is just more confusing.
In comparison to visibility, which is true in the default state, where each object is visible.

**Reply by belinski-roman** [reply]
> Для обсуждения @FreeCAD/design-working-group На мой взгляд, пункт меню со строкой «Переключить» всегда хуже, чем текущий дизайн, в котором вы мгновенно видите текущее состояние и можете переключаться между подавленным и не подавленным состояниями. Это обсуждалось, когда была представлена функция, а также для других команд (например, «Активная деталь/тело»). Я не вижу смысла устанавливать для подавленного состояния значение «истина», если по умолчанию оно равно «ложь», так как каждый объект по умолчанию не подавлен. Изменение метки на «Не подавлено» = «истина» только сбивает с толку. В отличие от видимости, которая является истинной в состоянии по умолчанию, когда каждый объект виден.

We can already see the current state in the tree. The suppressed element is crossed out.

**Reply by belinski-roman** [reply]
And with the example of Visibility, you contradict yourself. The variable for Visibility and Suppress should be similar, since they are similar entities.

**Comment by belinski-roman** [comment]
In the Linkstage branch, the menu item was dynamic (suppress/unsuppress). This is good because it allows you to work with multiple rows in the tree at the same time. The introduction of Toggle suppress along with this will make it even more versatile and understandable.
![Пример](https://github.com/user-attachments/assets/f59846d8-7fb8-4221-b2f8-ea0c5869b8be)

**Reply by belinski-roman** [reply]
This behavior can also be extended to all menu items with the Toggle prefix. But I'm still more concerned about the inverted value of the variable than the naming of the menu item.

**Reply by kadet1090** [reply]
Can you illustrate with example when such toggling of multiple (enabling one, disabling others) is useful?

Dynamic strings are possible but in general approach with checkbox is more common as you see from the menu the state.

**Reply by belinski-roman** [reply]
I rarely see checkboxes in the menu in other applications. The visualization of the state is already in the tree and it is always visible, unlike the checkbox in the menu.

I am responding to the application account - visual adjustment Part with configurations. Undoubtedly, it is more convenient to make such details using the configuration table or using VarSet (when it becomes possible). But if there are not many options, it will be possible to adjust each part manually, so to speak, using the visual method. Again, if you extend this behavior to all menu items containing the Toggle prefix, it will also be convenient, for example, for group freezing of parts in an assembly. 
An example from SolidWorks: 
![2024-12-27_01-03](https://github.com/user-attachments/assets/2e164c8e-0d92-413b-81f1-9fad697fd67b)

For some reason, everyone comments on changing the menu item, but the main message of this query is the inverted state of the variable. You can leave the menu alone, but change the default state to true.

**Reply by belinski-roman** [reply]
I did not write about the need to extend Suppress to other objects. I will create a separate Issue for this. I am sure that the Suppress function is also needed for Body, Parts, Assembly (Sub-assemblies), as well as in order to create assembly configurations with different parts. And for assemblies, the possibility of group manipulation by suppression/unsuppression is even more relevant.
